internal/terminal/table.go

```diff
@@ -0,0 +1,151 @@
+package terminal
+
+import (
+	"fmt"
+	"github.com/fatih/color"
+	"regexp"
+	"strings"
+)
+
+// table
+// headers is an ordered list of strings to specify the order of printing
+// data is a map of a string to another string to print, where the key should be a value in headers
+//	if not, we will not print
+// maxHeaderWidth is to determine the column width based on the max length of the values and header
+type table struct {
+	headers        []string
+	data           []map[string]string
+	maxHeaderWidth map[string]int
+}
+
+// columnBreakWidth is to set a standard for the break between columns for the table
+const (
+	 columnBreakWidth = 2
+)
+
+func newTable(headers []string, values []map[string]interface{}) *table {
+	if headers == nil || len(headers) == 0 ||
+		values == nil || len(values) == 0 {
+		return nil
+	}
+
+	maxHeaderWidth := make(map[string]int)
+	for _, h := range headers {
+		temp := len(h)
+		if temp > maxHeaderWidth[h] {
+			maxHeaderWidth[h] = temp
+		}
+	}
+
+	var stringValues []map[string]string
+	for _, v := range values {
+		tempVal := make(map[string]string)
+		for k1, v1 := range v {
+			// convert the interface values into strings for printing
+			var val string
+			switch v := v1.(type) {
+			case string:
+				spaceRegex := regexp.MustCompile(`\s+`)
+				val = spaceRegex.ReplaceAllString(v, " ")
```

Comment:
lets avoid any transformation logic for now and just set the raw `string` value (feel free to let me know I'm missing something and we need this now)

Reply:
If we don't replace any possible newlines, it breaks up the table formatting (\t, \n, etc) so I try to replace them here

Reply:
i see, that makes a lot of sense.  the one thing is I'm not sure we have/expect any table content to contain any newlines at this point.  similar to the `float`s, I'd rather remove this logic now and only add it if/when we need it (i.e. once the requirement for such logic is more defined/concrete)